### PR TITLE
[Prow CI Step Enablement] MGMT-12390: add step to enable MetalLB and Handling Ingress

### PR DESCRIPTION
**Pull request by LiangquanLi930**
This PR fixed ingress error
```
➜  oc get clusteroperators   
NAME                                       VERSION                         AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
console                                    4.13.0-0.ci-2023-06-16-161303   True        False         False      20m     
csi-snapshot-controller                    4.13.0-0.ci-2023-06-16-161303   True        False         False      114m    
dns                                        4.13.0-0.ci-2023-06-16-161303   True        False         False      71m     
image-registry                             4.13.0-0.ci-2023-06-16-161303   True        False         False      71m     
ingress                                    4.13.0-0.ci-2023-06-16-161303   True        False         False      113m    
insights                                   4.13.0-0.ci-2023-06-16-161303   True        False         False      72m     
kube-apiserver                             4.13.0-0.ci-2023-06-16-161303   True        False         False      114m    
kube-controller-manager                    4.13.0-0.ci-2023-06-16-161303   True        False         False      114m    
kube-scheduler                             4.13.0-0.ci-2023-06-16-161303   True        False         False      114m    
kube-storage-version-migrator              4.13.0-0.ci-2023-06-16-161303   True        False         False      71m     
monitoring                                 4.13.0-0.ci-2023-06-16-161303   True        False         False      70m     
network                                    4.13.0-0.ci-2023-06-16-161303   True        False         False      72m     
node-tuning                                4.13.0-0.ci-2023-06-16-161303   True        False         False      74m     
openshift-apiserver                        4.13.0-0.ci-2023-06-16-161303   True        False         False      114m    
openshift-controller-manager               4.13.0-0.ci-2023-06-16-161303   True        False         False      114m    
openshift-samples                          4.13.0-0.ci-2023-06-16-161303   True        False         False      70m     
operator-lifecycle-manager                 4.13.0-0.ci-2023-06-16-161303   True        False         False      114m    
operator-lifecycle-manager-catalog         4.13.0-0.ci-2023-06-16-161303   True        False         False      114m    
operator-lifecycle-manager-packageserver   4.13.0-0.ci-2023-06-16-161303   True        False         False      114m    
service-ca                                 4.13.0-0.ci-2023-06-16-161303   True        False         False      72m     
storage                                    4.13.0-0.ci-2023-06-16-161303   True        False         False      114m    
➜  oc get clusterversion version
NAME      VERSION                         AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.13.0-0.ci-2023-06-16-161303   True        False         20m     Cluster version is 4.13.0-0.ci-2023-06-16-161303
```

### CODE REVIEW

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by eranco74** [comment]
/lgtm
Do other platforms (kubevirt/ibm) run the full conformance tests suite?

File: ci-operator/step-registry/hypershift/agent/create/metallb/hypershift-agent-create-metallb-commands.sh [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  oc get csv "${CSV}" -n metallb-system -o yaml
+  echo
+  echo "csv ${CSV} Describe"
+  oc describe csv "${CSV}" -n metallb-system
+  exit 1
+fi
+
+echo "successfully installed metallb-operator"
+
+echo "config L2 Advertisement"
```
Comment (eranco74) [inline ci-operator/step-registry/hypershift/agent/create/metallb/hypershift-agent-create-metallb-commands.sh:79]: This echo statement is out of place

File: ci-operator/step-registry/hypershift/agent/create/metallb/hypershift-agent-create-metallb-commands.sh [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+oc create -f - <<EOF
+apiVersion: metallb.io/v1beta1
+kind: L2Advertisement
+metadata:
+  name: ingress-public-ip
+  namespace: metallb-system
+spec:
+  ipAddressPools:
+   - metallb
+EOF
```
Comment (eranco74) [inline ci-operator/step-registry/hypershift/agent/create/metallb/hypershift-agent-create-metallb-commands.sh:110]: Consider replacing it with BGPAdvertisement:
```
apiVersion: metallb.io/v1beta1
kind: BGPAdvertisement
metadata:
  name: ingress-public-ip
  namespace: metallb
spec:
  aggregationLength: 32128
  aggregationLengthV6: 128
EOF
```
We just updated the docs

**Comment by LiangquanLi930** [comment]
> /lgtm Do other platforms (kubevirt/ibm) run the full conformance tests suite?

yes, for kubevirt we have https://prow.ci.openshift.org/job-history/gs/origin-ci-test/logs/periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-kubevirt-baremetalds-conformance

File: ci-operator/step-registry/hypershift/agent/create/metallb/hypershift-agent-create-metallb-commands.sh (hunk 2 above)
Comment (LiangquanLi930) [reply ci-operator/step-registry/hypershift/agent/create/metallb/hypershift-agent-create-metallb-commands.sh:110]: > We just updated the docs

Could you help to share the link?
There is a small issue, does BGPAdvertisement not require setting ipAddressPools?

Comment (LiangquanLi930) [reply ci-operator/step-registry/hypershift/agent/create/metallb/hypershift-agent-create-metallb-commands.sh:110]: We just updated the docs

@eranco74 Could you help to share the link?
There is a small issue, does BGPAdvertisement not require setting ipAddressPools?

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Pull request merged**